lib/srv/db/access_test.go

```
@@ -255,114 +247,134 @@ func (m *testModules) Features() modules.Features {
+func (c *testContext) postgresClient(ctx context.Context, teleportUser, dbService, dbUser, dbName string) (*pgconn.PgConn, error) {
 	return postgres.MakeTestClient(ctx, common.TestClientConfig{
 		AuthClient: c.authClient,
 		AuthServer: c.authServer,
 		Address:    c.mux.DB().Addr().String(),
 		Cluster:    c.clusterName,
 		Username:   teleportUser,
 		RouteToDatabase: tlsca.RouteToDatabase{
-			ServiceName: c.postgresDBServer.GetName(),
+			ServiceName: dbService,
 			Protocol:    defaults.ProtocolPostgres,
 			Username:    dbUser,
 			Database:    dbName,
 		},
 	})
 }
 
-func (c *testContext) startHandlingMySQLConnections() {
-	// Start test MySQL server.
-	go c.mysqlServer.Serve()
-	// Start MySQL proxy server.
-	go c.proxyServer.ServeMySQL(c.mysqlListener)
-	// Start handling MySQL connections on the database server.
-	for conn := range c.proxyConn {
-		c.server.HandleConnection(conn)
-	}
-}
-
 // mysqlClient connects to test MySQL through database access as a specified
 // Teleport user and database account.
-func (c *testContext) mysqlClient(teleportUser, dbUser string) (*client.Conn, error) {
+func (c *testContext) mysqlClient(teleportUser, dbService, dbUser string) (*client.Conn, error) {
 	return mysql.MakeTestClient(common.TestClientConfig{
 		AuthClient: c.authClient,
 		AuthServer: c.authServer,
 		Address:    c.mysqlListener.Addr().String(),
 		Cluster:    c.clusterName,
 		Username:   teleportUser,
 		RouteToDatabase: tlsca.RouteToDatabase{
-			ServiceName: c.mysqlDBServer.GetName(),
+			ServiceName: dbService,
 			Protocol:    defaults.ProtocolMySQL,
 			Username:    dbUser,
 		},
 	})
 }
 
+// createUserAndRole creates Teleport user and role with specified names
+// and allowed database users/names properties.
+func (c *testContext) createUserAndRole(ctx context.Context, userName, roleName string, dbUsers, dbNames []string) (types.User, types.Role, error) {
```

optional: pass in `testing.T` and fail the test instead of returning an error